ethereum/core/src/test/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateArchiveTest.java

```diff
@@ -200,4 +201,58 @@ public void testGetMutableWithStorageConsistencyToRollbackAndRollForwardTheState
     verify(updater, times(1)).rollBack(any());
     verify(updater, times(1)).rollForward(any());
   }
+
+  @SuppressWarnings({"unchecked"})
+  @Test
+  public void testGetMutableWithRollbackNotOverrideTrieLogLayer() {
+    final KeyValueStorageTransaction keyValueStorageTransaction =
+        mock(KeyValueStorageTransaction.class);
+    when(keyValueStorage.startTransaction()).thenReturn(keyValueStorageTransaction);
+    final BlockHeader genesis = blockBuilder.number(0).buildHeader();
+    final BlockHeader blockHeaderChainA =
+        blockBuilder.number(1).timestamp(1).parentHash(genesis.getHash()).buildHeader();
+    final BlockHeader blockHeaderChainB =
+        blockBuilder.number(1).timestamp(2).parentHash(genesis.getHash()).buildHeader();
+
+    final Map<Bytes32, BonsaiLayeredWorldState> layeredWorldStatesByHash = mock(HashMap.class);
+    when(layeredWorldStatesByHash.containsKey(any(Bytes32.class))).thenReturn(true);
+    when(layeredWorldStatesByHash.get(eq(blockHeaderChainA.getHash())))
+        .thenReturn(mock(BonsaiLayeredWorldState.class, Answers.RETURNS_MOCKS));
+    when(layeredWorldStatesByHash.get(eq(blockHeaderChainB.getHash())))
+        .thenReturn(mock(BonsaiLayeredWorldState.class, Answers.RETURNS_MOCKS));
+
+    bonsaiWorldStateArchive =
+        spy(new BonsaiWorldStateArchive(storageProvider, blockchain, 12, layeredWorldStatesByHash));
+    var updater = spy(bonsaiWorldStateArchive.getUpdater());
+    when(bonsaiWorldStateArchive.getUpdater()).thenReturn(updater);
+
+    // initial persisted state hash key
+    when(blockchain.getBlockHeader(eq(Hash.ZERO))).thenReturn(Optional.of(blockHeaderChainA));
+    // fake trie log layer
+    final BytesValueRLPOutput rlpLogBlockA = new BytesValueRLPOutput();
+    final TrieLogLayer trieLogLayerBlockA = new TrieLogLayer();
+    trieLogLayerBlockA.setBlockHash(blockHeaderChainA.getHash());
+    trieLogLayerBlockA.writeTo(rlpLogBlockA);
+    final BytesValueRLPOutput rlpLogBlockB = new BytesValueRLPOutput();
+    final TrieLogLayer trieLogLayerBlockB = new TrieLogLayer();
+    trieLogLayerBlockB.setBlockHash(blockHeaderChainB.getHash());
+    trieLogLayerBlockB.writeTo(rlpLogBlockB);
+    when(keyValueStorage.get(blockHeaderChainA.getHash().toArrayUnsafe()))
```

Comment:
unit test failure is unnecessary mockito stubbings here

Reply:
fixed thanks